### ISSUE TITLE
Dungeon Crawl

### ISSUE THREAD

**Issue by TwinNumberOne** [siggame/MegaMinerAI-Game-Ideas#75]
**TLDR**
-
- Recruit for your guild 
- Build up your recruits, just be wary of getting snobbish!
- There are no labor laws and this town needs work for its unemployed population
- Fight to win, fight for money, and maybe if they pay you enough kill the boss
- the resources you harvest fight back (well not the ore or the herbs)
- being an adventurer in this universe sucks but it pays the bills

**Story**
-
After having been drawn to the area by what can only be described as a necromancer war two rival adventuring guilds open branch offices in the game town hoping to drain the local dungeon dry of money. However, the local town is destitute and the church needs to be bribed to reopen a local church. So the mayor of the town has set up a bidding contest to exploit the guild's competition and to see which guild can stay, the guilds must compete in their ability to harvest the dungeon for its resources of monster parts and ore where the most profitable guild can stay.

**The Game**
-
**Victory**
Make the most money by the major deadline it is that simple for each guild. This goes off the money and materials the player has stored at the end of the game so invest wisely.

**Map**
It is a multi-layer dungeon (you can just do teleportation points don't really need to add a 3rd axis). This muli layered business is done so the players can and should butt heads drawing away from each other hard-pulled monsters. At each side of the dungeon is an entrance and a guild where player units can drop off loot and be spawned in or upgraded by the player buying upgrades or recruiting more units. In the dungeon are veins of ore that can be mined by patsy, as well as patches of herbs that can be harvested for quick money or used to fully heal a unit on the spot.

**Movement**
each unit/mob has a move speed of 1-4 with 1 being super slow, 2 being slow, 3 being medium and 4 being fast.
Tanks/Patsy's can carry other incapacitated units but will be reduced to a movement speed of 1.
Move speed is also reduced by 1 stage when the unit is hit but only for a turn
A slowed unit is auto reduced to a movement speed of 1

**Resource Collection**
Monsters upon death leave goods on the spot they are killed this when brought back to base is converted to gold.
Ore that is mined by patsy by them standing next to a mineral patch and dumping the ore on the floor it can also be bought for money but it is used in some upgrades.
Herbs can be harvested from patches around the dungeon and can be used to heal units or sold for a good sum of money, they can also be bought too for a larger sum of money. Herb prices also drop for the player with less units than the other player (like with unit recruitment)
it should be noted that units cannot carry much loot so a patsy should always be better for moving large amounts of material around than the thief.

**Combat**
• For this due to the more complex nature of it dmg is simple as in all dmg is dmg, 
• when a player unit is dropped to 0 it is knocked out any more dmg kills it.
• Unit agro works via a unit getting too close to a mob or a mob being damaged which then agro's the units around it. The only way to ignore the proximity agro is with the sneaky ability.
• No matter what all units/mobs attack at the same speed, once a turn. Meaning most enemy mobs should not be too terribly threatening unless the player gets swarmed. So the name of the game is raid rules.
• Agro'ed mobs  get to act on their next turn (surprise), the player moves before the mobs

**Friendly Units**
Unit recruitment draws from a pool of available recruits the more money/units a player has than the other player the more arrogant their recruiter gets which makes the recruitment pool smaller. (upgraded units worth more than recruits I would say prob 3x more). The player gets recruits every couple of turns.
all unit costs are the cost to upgrade a recruit to the unit
units regenerate when at home base, or can be healed via herbs otherwise they do not heal.

1 = low
5 = high
cost = [gold/minerals]
| Name | Cost | Health |  Move Speed | attack | range | special |
|----|----|----|----|----|----|---|
|**Recuit**        |n/a|3|3|2|1|None|
|**Thief**          |4/1|1|5|3|5|S|
|**Tank**           |3/5|5|2|1|1|A/C|
|**Wizard**       |5/1|1|2|5|3|E|
|**Patsy**          |nope|2|3|1|2|A/B/C|

A - Expanded auto aggro zone around the unit (even nerfed their area is still larger than normal)
B - Can harvest materials
C - Carry and protect other units from physical attacks, as well as materials
E - AoE attacks
S - Does not draw agro unless enemy units are damaged, togglable for the unit,

**Enemy Mobs**
spawners just refill the dungeon and kill them nets a lot of gold but no more mobs to farm. Spawners also spawn an additional wave each time they are damaged. They also stop spawning when so many mobs are around them in a sense replenishing a room over time.

The price on the Necromancers head raises over time

0 = doesn't move
1 = low
5 = high
[+] = goes beyond saved for the boss and its miniboss

| Name | Health |  Move Speed | attack | range | special | 
|----|----|----|----|----|---|
|**Skeleton**             |2|5|3|1|None|
|**Zombie**               |5|2|3|1|None|
|**Vampire**              |3|3|3|5|None|
|**Slime**                   |3|3|2|3|E+/S/|
|**Spawner**              |5+|0|1|2|F/U+/Z|
|**Death Knight**        |5+|4|5|2|E/Z|
|**Sleepy Necromancer**     |5+|1|5|5|E+/F/Z+/U|

E - AoE attacks
F - regenerates over time
S - slows hit target
U - summons mobs
Z - triggers units around it as if it was hit if it is agro'ed in any way
[+] -Boosted ability 
[-] -Nerfed ability

**Remarks**
• My other idea was to make it so the basic recruit unit could just have its stats upgraded instead of a class system but I worried that while seeming to be simpler it would actually make things much more complicated for the player AI to understand as instead of a unit falling into a category to be understood each unit could fill any more of compiled roles or just be a uber unit that does everything.
• The enemy units in this aren't meant to be much of a challenge unless the player is swarmed and units start being drawn away from the tank to their backline. Though they are meant to slowly wear the player down
• It is possible for the enemy player to draw enemies to their opponent using their thief to setup ambushes and turn the field to their advantage to kill the other player's units to slow down their growth
• I was tempted to make it so wizards could do stuff like summon walls or set traps for the so they could do stuff like build kill boxes but that just seems rife for abuse though I suppose you could just do barriers where units are stopped as they cross them so they can just slow instead of cutting areas off. However, that might make wizards tempting to just spam and barrier slow the enemy while they slaughter the enemy. Aka less adventuring guild and more wizard school.
• I was also tempted to make it so they could place traps too but being high dmg aoe with low range and lower movement seems like they are more used to draw enemies in and slaughter them. Wherewith traps that could just get complicated to track and be used to try and abuse the enemy player AI and I don't want this to get any more complex.
• I was also tempted to add more enemies and like a dragon boss but that just started complicating things so I decided to reference Necrowar and make it so they are the same universe because that is fun.
• Was tempted to have healer units but making the player spend money to heal their units in the field makes it so they can only go so deep and have to retreat at some point or spend a lot of money. In short, it is to slow them down when clearing the dungeon.
• Overall the main thing about this is supposed to be unit tactics and proper positioning, with deeper play via the thief being used to draw enemy units to be killed or to go get lost loot.
• The purpose of the Z marker for the enemy units is to make it stuff like spawners or mini-bosses like the death knight if ever threatened to draw a sizable hoard to defend it and drive off the player making it so you can't just have your thief drawn a group off again and again.
• the point of slimes is to try and get the melee guys within range and or keep player units in range with vampires being the more dmg ranged units of the enemy again all to wear the player down over time.
• The unit recruitment pool, herbs price, and stipend are meant to in combo give a bounce back mechanic so a player can keep their units in the field for longer and more spam out more units. 
• I used to have it so you had to buy recruits but making them auto spawn in allows for more bounce back and I do think makes it a bit simpler as the programmer just has to worry about using what they get vs getting more units.
• was oh so tempted to add a mimic enemy but it just doesn't work in this scenario
• the patsy price is admittedly a bit of a dnd joke as then someone in the party is the designated patsy

**Comment by TwinNumberOne** [comment]
reworking to make less complex
